src/db/schema.ts

```diff
@@ -0,0 +1,288 @@
+  }),
+  owner: one(usersTable, {
+    fields: [tagOwnersTable.ownerId],
+    references: [usersTable.userId],
+  }),
+}));
+
+export const taggedRestaurantsTable = pgTable(
+  "taggedRestaurants",
+  {
+    tagId: uuid("tagId").references(() => tagsTable.tagId, {
+      onDelete: "cascade",
+      onUpdate: "cascade",
+    }),
+    placeId: varchar("placeId", { length: 300 }).references(
+      () => restaurantsTable.placeId,
+      {
+        onDelete: "cascade",
+        onUpdate: "cascade",
+      }
+    ),
+  },
+  (table) => ({
+    pk: primaryKey({ columns: [table.tagId, table.placeId] }),
+    tagIdIndex: index("tagIdIndex").on(table.tagId),
+  })
+);
+
+export const taggedRestaurantsRelations = relations(
+  taggedRestaurantsTable,
+  ({ one }) => ({
+    tag: one(tagsTable, {
+      fields: [taggedRestaurantsTable.tagId],
+      references: [tagsTable.tagId],
+    }),
+    restaurant: one(restaurantsTable, {
+      fields: [taggedRestaurantsTable.placeId],
+      references: [restaurantsTable.placeId],
+    }),
+  })
+);
+
+export const datesTable = pgTable(
+  "dates",
+  {
+    dateId: uuid("dateId").primaryKey().defaultRandom(),
+    occuredAt: timestamp("occuredAt").notNull(),
+  },
+  (table) => ({
+    dateIdAndOccuredAtIndex: index("dateIdAndOccuredAtIndex").on(
+      table.dateId,
+      table.occuredAt
+    ),
+  })
+);
+
+export const datesRelations = relations(datesTable, ({ many }) => ({
+  dateParticipantsTable: many(dateParticipantsTable),
+  privateMessagesTable: many(privateMessagesTable),
+}));
+
+export const dateParticipantsTable = pgTable(
+  "dateParticipants",
+  {
+    entryId: uuid("entryId").primaryKey().defaultRandom(),
+    dateId: uuid("dateId").references(() => datesTable.dateId, {
+      onDelete: "cascade",
+      onUpdate: "cascade",
+    }),
+    participantId: uuid("participantId").references(() => usersTable.userId, {
+      onDelete: "set null",
+      onUpdate: "cascade",
+    }),
+  },
+  (table) => ({
+    dateIdIndex: index("dateIdIndex").on(table.dateId),
+  })
+);
+
+export const dateParticipantsRelations = relations(
+  dateParticipantsTable,
+  ({ one }) => ({
+    date: one(datesTable, {
+      fields: [dateParticipantsTable.dateId],
+      references: [datesTable.dateId],
+    }),
+    participant: one(usersTable, {
+      fields: [dateParticipantsTable.participantId],
+      references: [usersTable.userId],
+    }),
+  })
+);
+
+export const privateMessagesTable = pgTable(
+  "privateMessages",
+  {
+    messageId: uuid("id").primaryKey().defaultRandom(),
+    dateId: uuid("dateId")
+      .notNull()
+      .references(() => datesTable.dateId, {
+        onDelete: "cascade",
+        onUpdate: "cascade",
+      }),
+    senderId: uuid("senderId").references(() => usersTable.userId, {
+      onDelete: "set null",
+      onUpdate: "cascade",
+    }),
+    sentAt: timestamp("sentAt").default(sql`now()`),
+    content: varchar("content", { length: 1000 }).notNull(),
```

Comment:
`varchar` -> `text`

Reply:
Should we not limit the length of a message to prevent a user from sending an infinitely-long message?

Reply:
hmm, maybe discuss in the future

Reply:
I changed it to text.